src/helpers/terrahub.js

```diff
@@ -105,11 +123,11 @@ class Terrahub {
       switch (extension) {
         case '.js':
           return () => {
-            const promise = require(args[0])(this._config, res);
+            const promise = require(args[0])(this._config, res.buffer);
 
-            return promise instanceof Promise ?
+            return (promise instanceof Promise ?
               promise :
-              Promise.resolve(promise);
+              Promise.resolve(promise)).then(() => Promise.resolve(res));
```

Comment:
It is not required to return `hook` result, because it is not used anywhere further
```suggestion
              Promise.resolve()).then(() => Promise.resolve(res));
```